### ISSUE TITLE
Shim is delayed by exactly 10 seconds

### ISSUE THREAD

**Issue by thexperiments** [chocolatey/home#197]
### What You Are Seeing?
Start of make (installed via chocolatey) is delayed by 10 seconds

### What is Expected?
Start of make is intant

### How Did You Get This To Happen? (Steps to Reproduce)

1. install make via chocolatey
2. execute make from comandline
Example make file
```
test:
	$(info this is an empty test)
```

### System Details

* OS Build (In PowerShell run `[System.Environment]::OSVersion.version.tostring()`): 10.0.19044.0
* Windows PowerShell version (Run: `$PSVersionTable`): 5.1.19041.1682
* Chocolatey version (Run `choco --version`): 1.1.0

### Output Log

~~~sh
make --shimgen-log test

### 10 second delay before first output

[shim]: Set up Shim to run with the following parameters:
  path to executable: C:\ProgramData\chocolatey\lib\make\tools\install\bin\make.exe
  working directory: c:\dev\test
  is gui? False
  wait for exit? True
  command (optional):
[shim]: Command line 'make  --shimgen-log test'
[shim]: Current process '"C:\ProgramData\chocolatey\bin\make.exe"'
[shim]: Command line after removing process 'make  --shimgen-log test'
[shim]: Shifting off the first argument since process didn't catch it
[shim]: Arguments after removing shimgen args - 'test'
[shim]: Arguments are 'test'
[shim]: Calling 'C:\ProgramData\chocolatey\lib\make\tools\install\bin\make.exe test'
this is an empty test
make: 'test' is up to date.
[shim]: Exiting with '0'.

~~~

timecmd.bat
```
@echo off
@setlocal

set start=%time%

:: Runs your command
cmd /c %*

set end=%time%
set options="tokens=1-4 delims=:.,"
for /f %options% %%a in ("%start%") do set start_h=%%a&set /a start_m=100%%b %% 100&set /a start_s=100%%c %% 100&set /a start_ms=100%%d %% 100
for /f %options% %%a in ("%end%") do set end_h=%%a&set /a end_m=100%%b %% 100&set /a end_s=100%%c %% 100&set /a end_ms=100%%d %% 100

set /a hours=%end_h%-%start_h%
set /a mins=%end_m%-%start_m%
set /a secs=%end_s%-%start_s%
set /a ms=%end_ms%-%start_ms%
if %ms% lss 0 set /a secs = %secs% - 1 & set /a ms = 100%ms%
if %secs% lss 0 set /a mins = %mins% - 1 & set /a secs = 60%secs%
if %mins% lss 0 set /a hours = %hours% - 1 & set /a mins = 60%mins%
if %hours% lss 0 set /a hours = 24%hours%
if 1%ms% lss 100 set ms=0%ms%

:: Mission accomplished
set /a totalsecs = %hours%*3600 + %mins%*60 + %secs%
echo command took %hours%:%mins%:%secs%.%ms% (%totalsecs%.%ms%s total)
```

**Comment by pauby** [comment]
Can you go back and update the issue description with the template?

**Comment by corbob** [comment]
To expand on what @pauby requested and hopefully avoid back and forth causing delays: A minimal reproducible example would be good. For instance including what `timecmd.bat` contains (I'm guessing it gets the time, runs the command, then compares the result, but just to be sure), as well as what directory structure and file contents are needed.

**Comment by thexperiments** [comment]
### What You Are Seeing?
Start of make (installed via chocolatey) is delayed by 10 seconds

### What is Expected?
Start of make is intant

### How Did You Get This To Happen? (Steps to Reproduce)

1. install make via chocolatey
2. execute make from comandline
Example make file
```
test:
	$(info this is an empty test)
```



### System Details

* OS Build (In PowerShell run `[System.Environment]::OSVersion.version.tostring()`): 10.0.19044.0
* Windows PowerShell version (Run: `$PSVersionTable`): 5.1.19041.1682
* Chocolatey version (Run `choco --version`): 1.1.0

### Output Log


~~~sh
make --shimgen-log test

### 10 second delay before first output

[shim]: Set up Shim to run with the following parameters:
  path to executable: C:\ProgramData\chocolatey\lib\make\tools\install\bin\make.exe
  working directory: c:\dev\test
  is gui? False
  wait for exit? True
  command (optional):
[shim]: Command line 'make  --shimgen-log test'
[shim]: Current process '"C:\ProgramData\chocolatey\bin\make.exe"'
[shim]: Command line after removing process 'make  --shimgen-log test'
[shim]: Shifting off the first argument since process didn't catch it
[shim]: Arguments after removing shimgen args - 'test'
[shim]: Arguments are 'test'
[shim]: Calling 'C:\ProgramData\chocolatey\lib\make\tools\install\bin\make.exe test'
this is an empty test
make: 'test' is up to date.
[shim]: Exiting with '0'.

~~~

timecmd.bat
```
@echo off
@setlocal

set start=%time%

:: Runs your command
cmd /c %*

set end=%time%
set options="tokens=1-4 delims=:.,"
for /f %options% %%a in ("%start%") do set start_h=%%a&set /a start_m=100%%b %% 100&set /a start_s=100%%c %% 100&set /a start_ms=100%%d %% 100
for /f %options% %%a in ("%end%") do set end_h=%%a&set /a end_m=100%%b %% 100&set /a end_s=100%%c %% 100&set /a end_ms=100%%d %% 100

set /a hours=%end_h%-%start_h%
set /a mins=%end_m%-%start_m%
set /a secs=%end_s%-%start_s%
set /a ms=%end_ms%-%start_ms%
if %ms% lss 0 set /a secs = %secs% - 1 & set /a ms = 100%ms%
if %secs% lss 0 set /a mins = %mins% - 1 & set /a secs = 60%secs%
if %mins% lss 0 set /a hours = %hours% - 1 & set /a mins = 60%mins%
if %hours% lss 0 set /a hours = 24%hours%
if 1%ms% lss 100 set ms=0%ms%

:: Mission accomplished
set /a totalsecs = %hours%*3600 + %mins%*60 + %secs%
echo command took %hours%:%mins%:%secs%.%ms% (%totalsecs%.%ms%s total)
```

**Comment by pauby** [comment]
@thexperiments I have updated the original comment with the one above and hidden the one above.

**Comment by corbob** [comment]
@thexperiments : thank you for the reproduction information. Unfortunately, I am unable to reproduce the results you're seeing:

![image](https://user-images.githubusercontent.com/30301021/196565695-1661f343-71f8-4308-b4be-e157dcb57bf3.png)

Thinking that maybe this was something solved in a newer shimgen and perhaps you installed Make with an old one I downgraded Chocolatey and reinstalled make. It shows a version of 0.8.1, but still doesn't exhibit this same issue. Could you possibly get the file version of your make.exe shim? (in PowerShell you can run `Get-ItemProperty <pathToFile>` and it'll be one of the properties. If it is older than 0.8.1, I would suggest trying to reinstall the package to have shimgen create a new shim and trying with that. 

![image](https://user-images.githubusercontent.com/30301021/196565802-6c7912ea-13f2-4d4e-adb0-262d82e3c27b.png)

**Comment by thexperiments** [comment]
Thanks for your efforts. Sadly (or luckily) I can't reproduce this anymore myself.

Current suspicion is that the delay was caused by crowdstrike endpoint protection but sadly this only is a suspicion and I can't proof it as it is not reproducible anymore. Current guess is that crowdstrike policies have been updated due to chocolatey interferences in other parts of the company.

However I have discovered that my system still shows similar delays with other applications and I'm continuing to investigate this.

If you know about any previous problems with crwdstrike endpoint protection it would be great to know and to prepare for the problems.

Thanks!
